lasagne/layers/helper.py

```diff
@@ -183,6 +189,13 @@ def get_output(layer_or_layers, inputs=None, **kwargs):
                                  "mapping this layer to an input expression."
                                  % layer)
             all_outputs[layer] = layer.get_output_for(layer_inputs, **kwargs)
+            if not isinstance(layer, Mock) and \
+                    not isinstance(layer.get_output_for, Mock):
```

Comment:
I'd prefer not to test for a `Mock` instance. What about `if isinstance(layer, Layer)` or `hasattr(layer, 'get_output_for')`?